cfme/tests/containers/test_configurable_menus.py

```diff
@@ -34,35 +16,26 @@ def is_menu_visible(appliance, link_text):
     return link_text in logged_in_page.navigation.nav_links()
 
 
-@pytest.yield_fixture(scope='module')
-def config_menus_visible(appliance):
-    config_menus_visibility(appliance, True)
-    yield
-    config_menus_visibility(appliance, False)
+@pytest.fixture(scope='function')
+def is_datawarehouse_menu_visible(appliance):
+    return is_menu_visible(appliance, 'Datawarehouse')
+
+
+@pytest.fixture(scope='function')
+def is_monitoring_menu_visible(appliance):
+    return is_menu_visible(appliance, 'Monitor')
 
 
 @pytest.mark.polarion('CMP-10614')
-def test_datawarehouse_invisible(appliance):
+def test_datawarehouse_invisible(is_datawarehouse_menu_visible):
     # This should be the default state
-    yaml_config = appliance.get_yaml_config()
-    assert not yaml_config['product']['datawarehouse_manager'], \
-        'Datawarehouse menu should be configured as invisible by default ' \
-        '(currently configured as visible)'
-    assert not is_menu_visible(appliance, 'Datawarehouse')
+    # Verifies BZ#1421175
+    assert not is_datawarehouse_menu_visible
 
 
+@pytest.mark.uncollectif(lambda: current_version() > "5.8")
 @pytest.mark.polarion('CMP-10613')
-def test_monitoring_invisible(appliance):
+def test_monitoring_invisible(is_monitoring_menu_visible):
     # This should be the default state
-    yaml_config = appliance.get_yaml_config()
-    assert not yaml_config['prototype']['monitoring'], \
-        'Monitor menu should be configured as invisible by default ' \
-        '(currently configured as visible)'
-    assert not is_menu_visible(appliance, 'Monitor')
-
-
-@pytest.mark.uncollectif(lambda: current_version() < "5.9")
-@pytest.mark.polarion('CMP-10649')
-def test_monitoring_visible(appliance, config_menus_visible):
-    assert is_menu_visible(appliance, 'Monitor'), \
-        'Monitor menu should be visible (currently invisible)'
+    # Verifies BZ#1421173
+    assert not is_monitoring_menu_visible
```

Comment:
**REQUIRED/QUESTION**: I don't think asserting a fixture variable is the way to go.

Reply:
I'm passing the returned value of the fixture which is whether the menu is visible or not (this is actually what the test tests). This is the best way to distinguish between Error/Failure.